tutorials/llm/llama-3/README.rst

```diff
@@ -0,0 +1,208 @@
+
+This is a demonstration of deploying multiple LoRA adapters with NVIDIA
+NIM. NIM supports LoRA adapters in ``.nemo`` (from NeMo Framework), and
+Hugging Face model formats. We will deploy the PubMedQA LoRA adapter
+from the first notebook, alongside two other already trained LoRA adapters
+(`GSM8K <https://github.com/openai/grade-school-math>`__,
+`SQuAD <https://rajpurkar.github.io/SQuAD-explorer/>`__) that are
+available on NVIDIA NGC as examples.
+
+``NOTE``: While it’s not necessary to complete the LoRA training and
+obtain the adapter from the previous notebook (“Creating a LoRA adapter
+with NeMo Framework”) to follow along with this one, it is recommended
+if possible. You can still learn about LoRA deployment with NIM using
+the other adapters downloaded from NGC.
+
+
+1. Download example LoRA adapters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following steps assume that you have authenticated with NGC and
+downloaded the CLI tool, as mentioned in pre-requisites.
+
+.. code:: bash
+
+   # Set path to your LoRA model store
+   export LOCAL_PEFT_DIRECTORY="./loras"
+   mkdir -p $LOCAL_PEFT_DIRECTORY
+   pushd $LOCAL_PEFT_DIRECTORY
+
+
+   cd $LORA_MODEL_STORE
+   # downloading NeMo-format loras
+   ngc registry model download-version "nim/meta/llama3-8b-instruct-lora:nemo-math-v1"
+   ngc registry model download-version "nim/meta/llama3-8b-instruct-lora:nemo-squad-v1"
+
+   popd
+   chmod -R 777 $LOCAL_PEFT_DIRECTORY
+
+2. Prepare the LoRA model store
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure that the the LoRA model store directory has a structure like so -
+with the name of the model as a sub-folder that contains the .nemo file.
+
+::
+
+   </path/to/LoRA-model-store>
+   ├── llama3-8b-instruct-lora_vnemo-math-v1
+   │   └── llama3_8b_math.nemo
+   ├── llama3-8b-instruct-lora_vnemo-squad-v1
+   │   └── llama3_8b_squad.nemo
+   └── llama3-8b-pubmed-qa
+       └── megatron_gpt_peft_lora_tuning.nemo
+
+The last one was just trained on the PubmedQA dataset in the previous
+notebook. After training is complete, that LoRA model checkpoint will be
+created at
+``./results/Meta-Llama-3-8B-Instruct/checkpoints/megatron_gpt_peft_lora_tuning.nemo``,
+assuming default paths in the first notebook weren’t modified.
+
+To ensure model store is organized as expected, create a folder named
+``llama3-8b-pubmed-qa``, and move your .nemo checkpoint there.
+
+.. code:: bash
+
+   mkdir -p $LOCAL_PEFT_DIRECTORY/llama3-8b-pubmed-qa
+
+   # Ensure the source path is correct
+   cp ./results/Meta-Llama-3-8B-Instruct/checkpoints/megatron_gpt_peft_lora_tuning.nemo $LOCAL_PEFT_DIRECTORY/llama3-8b-pubmed-qa 
+
+3. Set-up NIM
+^^^^^^^^^^^^^
+
+From your host OS environment, start the NIM docker container while
+mounting the LoRA model store, as follows:
+
+.. code:: bash
+
+   export NGC_API_KEY=<YOUR_NGC_API_KEY>
+   export LOCAL_PEFT_DIRECTORY=</path/to/LoRA-model-store>
+   chmod -R 777 $LOCAL_PEFT_DIRECTORY
```

Comment:
These two can be deleted since we did them up above

Reply:
removed